scripts/west_commands/runners/jlink.py

```diff
@@ -150,19 +150,7 @@ def jlink_version(self):
         # to load the shared library distributed with the tools, which
         # provides an API call for getting the version.
         if not hasattr(self, '_jlink_version'):
-            plat = sys.platform
-            if plat.startswith('win32'):
-                libname = Library.get_appropriate_windows_sdk_name() + '.dll'
-            elif plat.startswith('linux'):
-                libname = Library.JLINK_SDK_NAME + '.so'
-            elif plat.startswith('darwin'):
-                libname = Library.JLINK_SDK_NAME + '.dylib'
-            else:
-                self.logger.warning(f'unknown platform {plat}; assuming UNIX')
-                libname = Library.JLINK_SDK_NAME + '.so'
-
-            lib = Library(dllpath=os.fspath(Path(self.commander).parent /
-                                            libname))
+            lib = Library()
```

Comment:
actually, it looks like this would introduce a behavioural change -- it used to load the JLink DLL from the path specified by the `--commander` argument (or the default), but that is no longer the case after this change.

This may lead to `jlink_version` reporting an incorrect version (different from that of the "commander") on a system with multiple JLink installations (not so uncommon).

Reply:
fair point. changed strategy